src/DTIntrospection.cpp

```diff
@@ -0,0 +1,661 @@
+ * The supported operations are "=", "<", and ">".
+ *
+ * @param featureValueCombo A string representing the feature-value combination in the format "feature=op=value".
+ * @return A vector of integers representing the sample indices that match the given feature-value combination.
+ * @throws std::runtime_error If the feature-value syntax is incorrect.
+ */
+vector<int> DTIntrospection::getSamplesForFeatureValueCombo(string featureValueCombo) {
+    FeatureOpValue featureOpValue = extractFeatureOpValue(featureValueCombo);
+    double valueAsDouble = convert(featureOpValue.value);
+    vector<int> samples = {};
+
+    auto trainingDataDict = _dt->getTrainingDataDict();
+    auto featureNames = _dt->getFeatureNames();
+
+    if (featureOpValue.op == "=") {
+        for (const auto& samplePair : trainingDataDict) {
+            const int sample = samplePair.first;
+            vector<string> featuresAndValues = samplePair.second;
+
+            for (int i = 0; i < featuresAndValues.size(); i++) {
+                featuresAndValues[i].insert(0, featureNames[i] + "=");
+            }
+
+            if (std::find(featuresAndValues.begin(), featuresAndValues.end(), featureValueCombo) != featuresAndValues.end()) {
+                samples.push_back(sample);
+            }
+        }
+    }
+    else if (featureOpValue.op == "<") {
+        for (const auto& samplePair : trainingDataDict) {
+            const int sample = samplePair.first;
+            vector<string> featuresAndValues = samplePair.second;
+
+            for (int i = 0; i < featuresAndValues.size(); i++) {
+                featuresAndValues[i].insert(0, featureNames[i] + "=");
+            }
+            
+            for (const auto& item : featuresAndValues) {
+                FeatureOpValue featureOpValue2 = extractFeatureOpValue(item);
+                double value2AsDouble = convert(featureOpValue2.value);
+                
+                if (!std::isnan(valueAsDouble) && !std::isnan(value2AsDouble)) {
+                    if (featureOpValue.feature == featureOpValue2.feature && value2AsDouble <= valueAsDouble) {
+                        samples.push_back(sample);
+                        break;
+                    }
+                }
+
+            }
+            
+        }
+    }
+    else if (featureOpValue.op == ">") {
+        for (const auto& samplePair : trainingDataDict) {
+            const int sample = samplePair.first;
+            vector<string> featuresAndValues = samplePair.second;
+
+            for (int i = 0; i < featuresAndValues.size(); i++) {
+                featuresAndValues[i].insert(0, featureNames[i] + "=");
+            }
+            
+            for (const auto& item : featuresAndValues) {
+                FeatureOpValue featureOpValue2 = extractFeatureOpValue(item);
+                double value2AsDouble = convert(featureOpValue2.value);
+                
+                if (!std::isnan(valueAsDouble) && !std::isnan(value2AsDouble)) {
```

Comment:
Logic is duplicated, would be a good candidate for using a lambda or something in a templated function.

Reply:
Could be another optimization in the future, noted.